code/client/api.c

```diff
@@ -736,26 +736,27 @@ HQAPI BagEnum GetItemLocation(uint32_t item_id, unsigned int *slot)
     return bag;
 }
 
-HQAPI size_t GetItemModStruct(uint32_t item_id, uint32_t* buffer, size_t length)
+HQAPI int GetItemModStruct(uint32_t item_id, uint32_t* buffer, size_t length)
 {
     assert(client != NULL);
+    int written = -1;
```

Comment:
Do you need to differentiate between invalid item id and no mod?

If yes, I will change that to be:
```c
HQAPI bool GetItemModStruct(uint32_t item_id, uint32_t* buffer, size_t length, size_t *return_length)
```
If not, I will leave it as a `size_t` and return 0 when the item id isn't found.

Don't need to change your pr, I'll do it before merging.

Reply:
You can do it either way, but the usual approach to writing to a buffer would be to return the length, or -1 on failure

Reply:
Depend what you mean by the usual approach. The one that gave us decades of suffering? :D

I think it's better to have second approach, since it's harder to forgot, and it doesn't conflate the meaning of the return value. Also, It's not more lines of code.
```c
int ret = GetItemModStruct(...);
if (ret < 0)
    return err;
size_t n_mod = (size_t)ret;
```
where you would do something like:
```c
size_t n_mod
if (!GetItemModStruct(..., &n_mod))
    return err;
```